src/snowcli/snow_connector.py

```diff
@@ -35,15 +44,36 @@ def connect_to_snowflake(temporary_connection: bool = False, connection_name: Op
             {k: v for k, v in overrides.items() if v is not None}
         )
 
+    private_key = None
+    if "private_key_path" in connection_parameters:
+        if (
+            "authenticator" in connection_parameters
+            and connection_parameters["authenticator"] == "SNOWFLAKE_JWT"
+        ):
+            private_key_passphrase = os.getenv("PRIVATE_KEY_PASSPHRASE", None)
+            private_key = load_pem_to_der(connection_parameters["private_key_path"])
+            del connection_parameters["private_key_path"]
+        else:
+            raise Exception(
+                "Private Key authentication requires authenticator set to SNOWFLAKE_JWT"
+            )
+
     try:
         # Whatever output is generated when creating connection,
         # we don't want it in our output. This is particularly important
         # for cases when external browser and json format are used.
         with contextlib.redirect_stdout(None):
-            return snowflake.connector.connect(
-                application=_find_command_path(),
-                **connection_parameters,
-            )
+            if private_key is None:
+                return snowflake.connector.connect(
+                    application=_find_command_path(),
+                    **connection_parameters,
+                )
+            else:
+                return snowflake.connector.connect(
+                    application=_find_command_path(),
+                    private_key=private_key,
+                    **connection_parameters,
+                )
```

Comment:
I would suggest to do something like this:
```py
connection_parameters["private_key"] = private_key
with contextlib.redirect_stdout(None):
                return snowflake.connector.connect(
                    application=_find_command_path(),
                    **connection_parameters,
                )
```
WDYT?

Reply:
So, I tried this initially, but:

`connection_parameters["private_key"] = private_key`

Fails because private key is a bytes object.  Any thoughts on how I could fix that?

`_ConvertError: Invalid type <class 'bytes'>`